cpu/cpu.go

```diff
@@ -156,8 +156,15 @@ func (c *Cpu) memoryAddress(in Instruction) uint16 {
 		return uint16(in.Op8 + c.X)
 	case zeropageY:
 		return uint16(in.Op8 + c.Y)
+
+	// 65C02-only modes
+	case zpindirect:
+		return c.Bus.Read16(uint16(in.Op8))
+	case indirect:
+		return c.Bus.Read16(uint16(in.Op16))
+
 	default:
-		panic("unhandled addressing")
+		panic(fmt.Sprintf("unhandled addressing: %v", in.addressing))
```

Comment:
This is mostly for debugging. If one was going to report the addressing type properly, one would use the naming table.